docs/docker.rst

```diff
@@ -0,0 +1,162 @@
+
+    # First run will generate the rally.ini
+    docker run --rm -v $PWD/tmpdir:/rally/.rally elastic/rally --track=nyc_taxis --test-mode --pipeline=benchmark-only --target-hosts=es01:9200
+
+        ____        ____
+       / __ \____ _/ / /_  __
+      / /_/ / __ `/ / / / / /
+     / _, _/ /_/ / / / /_/ /
+    /_/ |_|\__,_/_/_/\__, /
+                    /____/
+
+    Running simple configuration. Run the advanced configuration with:
+
+      esrally configure --advanced-config
+
+    * Setting up benchmark root directory in /rally/.rally/benchmarks
+    * Setting up benchmark source directory in /rally/.rally/benchmarks/src/elasticsearch
+
+    Configuration successfully written to /rally/.rally/rally.ini. Happy benchmarking!
+
+    More info about Rally:
+
+    * Type esrally --help
+    * Read the documentation at https://esrally.readthedocs.io/en/latest/
+    * Ask a question on the forum at https://discuss.elastic.co/c/elasticsearch/rally
+
+    # now run our benchmark
+    docker run --rm -v $PWD/tmpdir:/rally/.rally elastic/rally --track=nyc_taxis --test-mode --pipeline=benchmark-only --target-hosts=es01:9200
+
+    ...
+
+    # inspect results
+    $ tree myrally/benchmarks/races/
+    myrally/benchmarks/races/
+    └── 2019-06-05-14-03-44
+        └── race.json
+
+    1 directory, 1 file
+
+
+In case you forgot to bind mount a directory, the Rally Docker image will create an `anonymous volume <https://success.docker.com/article/different-types-of-volumes>`_ for ``/rally/.rally`` to ensure logs and results get persisted even after the container has terminated.
+
+For example, after executing our earlier quickstart example ``docker run elastic/rally --track=nyc_taxis --test-mode --pipeline=benchmark-only --target-hosts=es01:9200``, ``docker volume ls`` shows a volume:::
+
+    $ docker volume ls
+    DRIVER              VOLUME NAME
+    local               96256462c3a1f61120443e6d69d9cb0091b28a02234318bdabc52b6801972199
+
+
+To further examine the contents we can bind mount it from another image e.g.::
+
+    $ docker run --rm -i -v=96256462c3a1f61120443e6d69d9cb0091b28a02234318bdabc52b6801972199:/rallyvolume -ti python:3.7.3-slim /bin/bash
+    root@9a7dd7b3d8df:/# cd /rallyvolume/
+    root@9a7dd7b3d8df:/rallyvolume# ls
+    root@9a7dd7b3d8df:/rallyvolume/.rally# ls
+    benchmarks  logging.json  logs	rally.ini
+    # head -4 benchmarks/races/2019-06-05-13-51-20/race.json
+    {
+     "rally-version": "1.2.0.dev0",
+     "environment": "local",
+     "trial-id": "1d81930a-4ebe-4640-a09b-3055174bce43",
+
+Specifics about the image
+-------------------------
+
+Rally runs as user ``1000`` and its files are installed with uid:gid ``1000:0`` (to support `Arbitrary User IDs <https://docs.openshift.com/enterprise/3.1/creating_images/guidelines.html>`_).
+
+Extending the Docker image
+--------------------------
+
+Creating your own customized Docker image is simple. You can start with a ``Dockerfile`` as simple as::
```

Comment:
We should probably rephrase this to avoid calling it "simple"? ([rationale](https://developers.google.com/style/tone)). How about this?

> You can also create your own customized Docker image on top of the existing one. The example below shows how to get started::

Reply:
Addressed in https://github.com/elastic/rally/pull/702/commits/af1a12402329af7bf27343bc33d63d3fef7abd4b